src/main/java/modify_event_use_case/ModifyEventDataAccess.java

```diff
@@ -53,12 +53,12 @@ public boolean titleExistsInDay(int dayIndex, String title) {
      */
     @Override
     public boolean isTimeConflict(int dayIndex, String title, LocalTime newStart, LocalTime newEnd) {
-        HashMap<String, DataAccessEvent> eventsToCheck = days.get(dayIndex).getEvents();
+        HashMap<String, Event> eventsToCheck = days.get(dayIndex).getEvents();
         for(String name: eventsToCheck.keySet()){
             if(!Objects.equals(name, title)){
                 LocalTime start = eventsToCheck.get(name).getStartTime();
                 LocalTime end = eventsToCheck.get(name).getEndTime();
-                if(!(newStart.isAfter(end) || newEnd.isBefore(start))){
+                if(newStart.isBefore(end) &&  newEnd.isAfter(start)){
```

Comment:
Upon further consideration, the if statement should be:
!(startTime.isBefore(end) && (endTime.isBefore(start) || endTime.equals(start))) &&
            !(start.isBefore(endTime) && (end.isBefore(startTime) || end.equals(startTime)))

Reply:
And can you make the same change to CreateEventDataAccess?

Reply:
the statement you suggested can be simplified a bit due to us checking for chronological order on start and end times.

Reply:
After writing out a derivation, I can affirm that the statement currently in the file and the statement being suggested are logically equivalent.

Reply:
Nevermind it was fine.